src/twisted/web/microdom.py

```diff
@@ -367,9 +366,9 @@ class Comment(CharacterData):
     def writexml(self, stream, indent='', addindent='', newl='', strip=0,
                  nsprefixes={}, namespace=''):
         val=self.data
-        if isinstance(val, UnicodeType):
-            val=val.encode('utf8')
-        stream.write("<!--%s-->" % val)
+        if isinstance(val, bytes):
+            val = val.decode('utf8')
+        stream.write(u"<!--{}-->".format(val))
```

Comment:
There is a minor interface consistency issue here; previously, this `writexml` method would have accepted an `io.BytesIO`; now, it requires a `io.StringIO`.  Normally I'd say that we should follow the example of the standard library, and their precedent says that `writexml` should be in terms of native `str`s and therefore text going forward.

However, *the only reason that this module exists at all* is the capricious interface changes in the stdlib's XML support that broke almost every XML-based application over the years, so interface stability is a particularly sore point here :).  Can you fix this to be permissive, and use `twisted.python.compat.ioType` to allow for either str- or bytes-based file-like objects?